src/garage/torch/algos/vpg.py

```diff
@@ -183,39 +225,60 @@ def _compute_loss(self, itr, obs, actions, rewards, valids, baselines):
             torch.Tensor: Calculated mean value of loss
 
         """
-        del itr
+        obs_flat = torch.cat(filter_valids(obs, valids))
+        actions_flat = torch.cat(filter_valids(actions, valids))
+        rewards_flat = torch.cat(filter_valids(rewards, valids))
+        advantages_flat = self._compute_advantage(rewards, valids, baselines)
 
-        policy_entropies = self._compute_policy_entropy(obs)
+        return self._compute_loss(obs_flat, actions_flat, rewards_flat,
+                                  advantages_flat)
 
-        if self._maximum_entropy:
-            rewards += self._policy_ent_coeff * policy_entropies
+    def _compute_loss(self, obs, actions, rewards, advantages):
+        """Compute mean value of loss.
+
+        Args:
+            obs (torch.Tensor): Observation from the environment.
+            actions (torch.Tensor): Predicted action.
+            rewards (torch.Tensor): Feedback from the environment.
+            advantages (torch.Tensor): Expected rewards over the actions.
+
+        Returns:
+            torch.Tensor: Calculated mean value of loss
```

Comment:
Please note this is a scalar.

Reply:
Do you mean like this?

`torch.Tensor: Calculated mean scalar value of loss`

Reply:
sure